neoantigenPredictor.wdl

```diff
@@ -198,71 +198,125 @@ task extractHLAs{
   
-  command<<<
-  python3 <<CODE
-  import pandas as pd
-  
-  files="~{sep=" " hlafiles}".split()
-  callers="~{sep=" " hlacallers}".split()
-  
-  ### this code will extract top HLAs from the various outputs, and store in a common format
-  hlas=[]
-  for file in files:
-    caller=callers.pop(0)
+command <<<
+set -e
+
+python3 - <<CODE
+import pandas as pd
+import re
+
+files = "~{sep=" " hlafiles}".split()
+callers = "~{sep=" " hlacallers}".split()
+
+# Regex patterns for allele validation
+t1k_pattern = re.compile(r'^HLA-[ABC]\*\d{2}:\d{2}(:\d{2,3}){0,2}[A-Z]*$')
+optitype_pattern = re.compile(r'^[ABC]\*\d{2}:\d{2}(:\d{2,3}){0,2}[A-Z]*$')
+
+### this code will extract top HLAs from the various outputs, and store in a common format
+all_hlas = []
+
+for file, caller in zip(files, callers):
+
     if caller == "t1k":
-      ### use the top 3 alleles (A,B,C), and show the HLAGene Family (col1, 5th character, A,B or C), and the first and second HLA Allele names (col 3,6)
-      with open(file) as f:
-        lines=f.readlines()[0:3]
-        for line in lines:
-          fields=line.replace("*","").split()
-          ## get the gene HLA-GENE* as a single character, A,B or C, a class I HLA
-          hla_gene=fields[0][4:]
-    
-          ## limit HLA codes to Field 1 and 2, removing everything after (split on :, then join the first two fields)
-          allele1=":".join(fields[2].split(":")[0:2])
-          allele2=":".join(fields[5].split(":")[0:2])
-          hlas.append([hla_gene,allele1])
-          hlas.append([hla_gene,allele2])
+        # For t1k output, use only the top 3 lines (corresponding to HLA-A, -B, -C) 
+        with open(file) as f:
+            lines = f.readlines()[0:3]
+            for line in lines:
+                fields = line.strip().split()
+
+                # Only process genes A, B, or C
+                gene_name = fields[0]
+                if gene_name not in ["HLA-A", "HLA-B", "HLA-C"]:
+                    continue
+
+                # Extract both alleles from the line
+                allele1_raw = fields[2]
+                allele2_raw = fields[5]
+
+                # Validate and extract allele1
+                if t1k_pattern.match(allele1_raw):
+                    parts = allele1_raw.split("*")
+                    gene = parts[0].replace("HLA-", "")
+                    allele = ":".join(parts[1].split(":")[0:2])
+                    all_hlas.append([gene, f"HLA-{gene}{allele}"])
+                else:
+                    print(f"Invalid allele1 format: {allele1_raw}")
+
+                # Validate and extract allele2
+                if t1k_pattern.match(allele2_raw):
+                    parts = allele2_raw.split("*")
+                    gene = parts[0].replace("HLA-", "")
+                    allele = ":".join(parts[1].split(":")[0:2])
+                    all_hlas.append([gene, f"HLA-{gene}{allele}"])
+                else:
+                    print(f"Invalid allele2 format: {allele2_raw}")
```

Comment:
minor comment: can you loop through these instead of repeating the same code for allele1 and allele2?

alleles_raw = [ fields[2], fields[3]]
for allele_raw in alleles_raw:
....